pkg/messageprocessors/javascript/javascript.go

```diff
@@ -205,21 +278,72 @@ func (h *host) decodeUplink(ctx context.Context, ids *ttnpb.EndDeviceIdentifiers
 		return err
 	}
 
-	var output decodeUplinkOutput
+	var output uplinkDecoderOutput
 	err = valueAs(&output)
 	if err != nil {
 		return errOutput.WithCause(err)
 	}
-	if len(output.Errors) > 0 {
-		return errOutputErrors.WithAttributes("errors", strings.Join(output.Errors, ", "))
-	}
 
-	s, err := gogoproto.Struct(output.Data)
+	if errs := output.Decoded.Errors; len(errs) > 0 {
+		return errOutputErrors.WithAttributes("errors", strings.Join(errs, ", "))
+	}
+	decodedPayload, err := gogoproto.Struct(output.Decoded.Data)
 	if err != nil {
 		return errOutput.WithCause(err)
 	}
-	msg.DecodedPayload = s
-	msg.DecodedPayloadWarnings = output.Warnings
+	msg.DecodedPayload, msg.DecodedPayloadWarnings = decodedPayload, output.Decoded.Warnings
+
+	if normalized := output.Normalized; normalized != nil {
+		if errs := normalized.Errors; len(errs) > 0 {
+			return errOutputErrors.WithAttributes("errors", strings.Join(errs, ", "))
+		}
+		if normalized.Data == nil {
+			return errOutput.New()
+		}
+		// The returned data can be an array of measurements or a single measurement object.
+		var ms []map[string]interface{}
```

Comment:
Pardon my nitpicking, but I'm curious, why do you prefer a short variable name for this one that has a lifespan of about 20 lines? It seems more readable to me to use `measurments` here and change `measurment` to `m` or something like that, since the later ones have a shorter scope.